### ISSUE TITLE
Proposal - Capture the X-Api-Info-Location header on all broker requests

### ISSUE THREAD

**Issue by vorbidan** [pivotal-cf/brokerapi#66]
## What
Add `infoLocation` to HTTP context. This will allow to obtain the foundation details if required.

## How
_Code untested_
Example of `brokerapi/middlewares/api-info-location-header.go`

```go
const (
	infoLocationKey = "infoLocation"
)

func AddInfoLocationToContext(next http.Handler) http.Handler {
	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
		infoLocation := req.Header.Get("X-Api-Info-Location")
		newCtx := context.WithValue(req.Context(), infoLocationKey, infoLocation)
		next.ServeHTTP(w, req.WithContext(newCtx))
	})
}
```
Update `api` to add MiddlewareFunc to the chain:
```go
[...]
router.Use(middlewares.AddInfoLocationToContext)
[...]
```

**Comment by winnab** [comment]
Thanks @vorbidan! For prioritization, can you give a bit more detail about how and how soon you are hoping to use this? The `X-Api-Info-Location` was removed from the OSBAPI spec, so it would be great to hear more about the reason for brokerapi to start accepting it.

**Comment by vorbidan** [comment]
Thank you for your reply @winnab, I was not aware that it has been specifically removed from the OSBAPI spec, even though I could not find it there... I found the discussion about removing the "CFisms" from the OSBAPI, but could not really link it to what came out of it specifically for the `X-Api-Info-location` 
Coming from the Java world and Spring Cloud Open Service Broker and the fact that it is still a part of the implementation in the latest release, I missed it in the Go ...
How would one "know" what platform is invoking the Service Broker? Should it be a part of the `context`?

**Comment by winnab** [comment]
It seems that, according to the spec, it's possible to identify the "type" of the originating platform by inspecting either the `X-Broker-API-Originating-Identity` or the request `context.platform`. That would inform your broker that the platform sending the request is a "cloudfoundry" platform.

Is that enough? Or are you looking for ways to ping back the platform or differentiate between multiple CFs?

**Comment by vorbidan** [comment]
The idea is to be able to ping back the platform.

**Comment by winnab** [comment]
Ok, makes sense. Would you like to do a pull request for this work? We'd be happy to review.

**Comment by vorbidan** [comment]
Done.
https://github.com/pivotal-cf/brokerapi/pull/67

**Comment by kirederik** [comment]
Thanks!

It is not available on v5.1.0
